tasks/ast_codegen/src/main.rs

```diff
@@ -0,0 +1,330 @@
+use std::{borrow::Cow, collections::HashMap, error::Error, io::Read, path::PathBuf};
+
+use itertools::Itertools;
+use syn::{parse_file, Ident};
+
+#[derive(Default)]
+struct AstCodegen {
+    files: Vec<PathBuf>,
+}
+
+type Result<R> = std::result::Result<R, String>;
+
+impl AstCodegen {
+    #[must_use]
+    fn add_file<P>(mut self, path: P) -> Self
+    where
+        P: AsRef<str>,
+    {
+        self.files.push(path.as_ref().into());
+        self
+    }
+
+    fn generate(self) -> Result<String> {
+        let mut schema = self
+            .files
+            .into_iter()
+            .map(Module::from)
+            .flat_map(Module::load)
+            .map(Module::expand)
+            .collect::<Result<Vec<_>>>()?;
+        let mut ty_table: HashMap<String, &mut AstType> = schema
+            .iter_mut()
+            .flat_map(|it| it.items.iter_mut())
+            .map(|it| (it.ident().to_string(), it))
+            .collect();
+        for (ident, ty) in &mut ty_table {
+            dbg!(ident, ty);
+        }
+        // dbg!(file_mod.keys());
+        // .map_ok(Module::build)
+        // .flatten()
+        // .collect();
+
+        // for s in schema.definitions {}
+
+        Ok(String::new())
+    }
+}
+
+use analyzer::*;
+use schema::*;
+mod schema {
```

Comment:
@overlookmotel this is what I want, can you double confirm this is what you want as well?

Reply:
Please see below...

Reply:
rzvxa and I have discussed. In my opinion this PR is good as first step. We can iterate on it later.